src/pages/home/Home.tsx

```diff
@@ -1,168 +1,163 @@
-  const [deleteStudent] = useDeleteStudentMutation();
   const navigate = useNavigate();
-  const [filter, setFilter] = useState("");
-  const [confirmDialogOpen, setConfirmDialogOpen] = useState(false);
-  const [studentToDelete, setStudentToDelete] = useState<{
-    id: string;
-    name: string;
-  } | null>(null);
-
-  const {
-    data: students,
-    error,
-    isLoading,
-  } = useGetUserStudentsQuery(userId ? { userId, filter } : skipToken);
+  const [studentList, setStudentList] = useState<StudentResponse[]>([]);
+  const [filter, setFilter] = useState<string>("");
+  const { data: students } = useGetUserStudentsQuery(
+    userId ? { userId, filter } : skipToken
+  );
+  useEffect(() => {
+    if (students) {
+      setStudentList(students);
+    }
+  }, [students]);
+  const sortedStudents = useMemo(() => {
+    return [...studentList].sort((a, b) => {
+      if (a.isActive !== b.isActive) return a.isActive ? -1 : 1;
+      return 0;
+    });
+  }, [studentList]);
+  console.log("Student List:", studentList);
+  const sortByField = (field: keyof StudentResponse, asc: boolean) => {
+    const sorted = [...studentList].sort((a, b) => {
+      const valueA = a[field];
+      const valueB = b[field];
 
-  if (isLoading) return <p>Cargando...</p>;
-  if (error) return <p>Error al cargar estudiantes</p>;
+      if (field === "dni") {
+        return asc
+          ? Number(valueA) - Number(valueB)
+          : Number(valueB) - Number(valueA);
+      }
+      return asc
+        ? String(valueA).localeCompare(String(valueB))
+        : String(valueB).localeCompare(String(valueA));
+    });
 
-  const handleDelete = async (studentId: string, studentName: string) => {
-    setStudentToDelete({ id: studentId, name: studentName });
-    setConfirmDialogOpen(true);
-  };
-  const handleConfirmDeleteClick = () => {
-    if (!studentToDelete) return;
-    deleteStudent(studentToDelete.id)
-      .unwrap()
-      .then(() => {
-        setConfirmDialogOpen(false);
-        setStudentToDelete(null);
-      });
-  };
-  const handleCancelDeleteClick = () => {
-    setConfirmDialogOpen(false);
-    setStudentToDelete(null);
+    setStudentList(sorted);
   };
+  const columns: Column<StudentResponse>[] = [
+    {
+      header: (
+        <SortableButton text="DNI" onSort={(asc) => sortByField("dni", asc)} />
+      ),
+      accessor: "dni",
+    },
+
+    {
+      header: (
+        <SortableButton
+          text="Nombre"
+          onSort={(asc) => sortByField("name", asc)}
+        />
+      ),
+      accessor: "name",
+    },
+    {
+      header: (
+        <SortableButton
+          text="Apellido"
+          onSort={(asc) => sortByField("lastname", asc)}
+        />
+      ),
+      accessor: "lastname",
+    },
+    {
+      header: "Situación",
+      render: (student: StudentResponse) => (
+        <SituationText status={student.status ? "Al día" : "Debe"}>
+          {student.status ? "Al día" : "Debe"}
+        </SituationText>
```

Comment:
Acá no deberías decidir qué enviar segun el valor del status del estudiante, sino que directamente enviar la información del status del estudiante


```suggestion
        <SituationText status={student.status}>
          {student.status}
        </SituationText>
```